src/strigo/index.ts

```
@@ -4,38 +4,31 @@ import * as urlTools from "../modules/url/url";
 import * as configManager from "../modules/config/config";
 import * as sessionManager from "../modules/session/session";
 import * as eventsStorageManager from "../modules/events-storage/events-storage";
-import iframeWidget from "../modules/widgets/iframe";
-import overlayWidget from "../modules/widgets/overlay";
 import { Logger } from "../../services/logger";
-
-import { WIDGET_FLAVORS } from "../modules/session/session.types";
 import { StrigoToken } from "../modules/config/config.types";
+import * as widgetFactory from "../modules/widgets/widget-factory";
 
 export namespace Strigo {
   export let SDKType: SDK_TYPES;
 
   export function init() {
-    Logger.info("init called");
+    Logger.info("Started initialization");
     eventsStorageManager.init();
+
     if (window.Strigo?.initialized) {
       return;
     }
     window.Strigo.initialized = true;
 
-    // Check if other instances exists
-    switch (sessionManager.getWidgetFlavor()) {
-      case WIDGET_FLAVORS.IFRAME: {
-        SDKType = iframeWidget.init();
-        break;
-      }
-      case WIDGET_FLAVORS.OVERLAY: {
-        SDKType = SDK_TYPES.OVERLAY;
-        overlayWidget.init();
-        break;
-      }
-      default:
-        break;
+    const widget = widgetFactory.getWidget(sessionManager.getWidgetFlavor());
+
+    if (!widget) {
+      Logger.error(`Can't finish initialization. widgetFlavor is not supported.`);
```

Possibly throw an error in getWidget function instead of logging the error